src/main/java/softdesign/Robot.java

```diff
@@ -35,223 +57,72 @@
-	private void checkDirection(CameraSensor camera, CameraSensor camera2, CameraSensor camera3) {
-		switch (currentDirection) {
-			case NORTH:
-				checkCameraInDirections(Direction.WEST, Direction.EAST, Direction.SOUTH);
-				break;
-			case SOUTH:
-				checkCameraInDirections(Direction.EAST, Direction.WEST, Direction.NORTH);
-				break;
-			case WEST:
-				checkCameraInDirections(Direction.SOUTH, Direction.NORTH, Direction.EAST);
-				break;
-			case EAST:
-				checkCameraInDirections(Direction.NORTH, Direction.SOUTH, Direction.WEST);
-				break;
-		}
-	}
-
-	//checks if the points left, right, and back in relation to the robot has been taken a picture of, if an image has not been taken, then an image will be taken
-	private void checkCameraInDirections(Direction left, Direction right, Direction back) {
-		if (isUnvisited(hasPointVisited(left))) {
-			coverAndTrack(this.camera, hasPointVisited(left));
-		}
-		if (isUnvisited(hasPointVisited(right))) {
-			coverAndTrack(this.camera2, hasPointVisited(right));
-		}
-		if (isUnvisited(hasPointVisited(back))) {
-			coverAndTrack(this.camera3, hasPointVisited(back));
-		}
-	}
-
-	// this method returns a new coordinate
-	private Point3d hasPointVisited(Direction direction) {
-		Point3d result;
-
-		switch (direction) {
-			case EAST:
-				result = new Point3d(coordinate.x, coordinate.y, coordinate.z - 1);
-				break;
-			case WEST:
-				result = new Point3d(coordinate.x, coordinate.y, coordinate.z + 1);
-				break;
-			case SOUTH:
-				result = new Point3d(coordinate.x + 1, coordinate.y, coordinate.z);
-				break;
-			case NORTH:
-				result = new Point3d(coordinate.x - 1, coordinate.y, coordinate.z);
-				break;
-			default:
-				throw new IllegalArgumentException(String.format(UNRECOGNIZED_DIRECTION_EXCEPTION, direction));
+		if (wallIsAhead() || visitedIsAhead()) {
+			setTranslationalVelocity(0);
+			rotateY(-(Math.PI / 2));
+			currentDirection = currentDirection.rightOf(1);
+		} else {
+			setTranslationalVelocity(1);
 		}
-
-		return result;
 	}
 
-	//takes picture of point and marks the point as visited
-	private void coverAndTrack(CameraSensor camera, Point3d coord) {
-		camera.copyVisionImage(cameraImage);
-		isVisited(coord);
+	//takes images from the back, left, and right side if not take yet
+	private void ensureNeighbouringImagesTaken() {
+		takeImageIfNeeded(currentDirection.rightOf(1), backCamera);
+		takeImageIfNeeded(currentDirection.rightOf(2), rightCamera);
+		takeImageIfNeeded(currentDirection.rightOf(3), leftCamera);
 	}
 
-	// returns if the coordinate two points in front of the currentDirection has been visited
-	private boolean isNearCovered() {
-		boolean result;
+	private void takeImageIfNeeded(Direction direction, CameraSensor camera) {
+		Point3d coord = stepsAhead(direction, 1);
```

Comment:
Can you rename this to `coordinate`? Rest of the code also uses `coordinate` and it is considered bad practice to use non-standard abbreviations. (Simbad is not a good example to follow)